app/modules/rfswitch.c

```diff
@@ -76,19 +98,25 @@ void transmit(HighLow pulses, bool invertedSignal, int pulseLength, int pin) {
  */
 void send(unsigned long protocol_id, unsigned long pulse_length, unsigned long repeat, unsigned long pin, unsigned long value, unsigned int length) {
   platform_gpio_mode(pin, PLATFORM_GPIO_OUTPUT, PLATFORM_GPIO_FLOAT);
-  Protocol p = proto[protocol_id-1];
+  Protocol p = custom;
+  if(protocol_id > 0) p = proto[protocol_id-1];
```

Comment:
Can you add a check for having the protocol id in range somewhere? It should probably throw a lua error if it is out of range. Otherwise I suspect that rfswitch.send(9999999, ...) will cause the system to fault.